src/Xamarin.Android.Build.Tasks/Tasks/ResolveJdkJvmPath.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.Build.Framework;
```

Comment:
Let ask @jonpryor about that, as we already had the separate task once https://github.com/xamarin/xamarin-android/pull/2153#pullrequestreview-153344730 and he suggested to do it in `ResolveSdks` instead.

Now the circumstances changed (the DTB performance is important), but I guess we can still keep it in `ResolveSdks` and add `ResolveJvmPath` *bool* parameter to specify, whether the task should also resolve the `JvmPath`? That parameter value would be: `'$(AndroidGenerateJniMarshalMethods)' == 'True'`

I am OK with both alternatives, just don't remember what was the reasoning for not having it in separate task.

Reply:
My original concern was around *consistency* of the JVM path which was used:

> This way we ensure that we're consistent.  By using `JdkInfo.GetKnownSystemJdkInfos()`, it's possible that `<ResolveSdks/>` will provide one JDK path, but `JdkJvmPath` could come from a *different* path. 

I *believe* that in the original incarnation of the `<JdkInfo/>` task added in PR #2153, `<JdkInfo/>` didn't accept an "input" Java SDK Path property, but simply *re-* computed the JVM path based on `Xamarin.Android.Tools.JdkInfo`, which might not find the same path `<ResolveSdks/>` would have come up with (in part because `<ResolveSdks/>` reads `$(JavaSdkDirectory)`, which could be overridden to point to ~anywhere).

Keeping both the `JavaSdkPath` and `JdkJvmPath` logic within `<ResolveSdks/>` thus helped ensur a degree of consistency.

The approach here *does* afford consistency between the `<ResolveJdkJvmPath/>` and `<ResolveSdks/>` tasks, because of the `ResolveJdkJvmPath.JavaSdkPath` property, and the only way they could differ is if the path that `<ResolveSdks/>` computes is an "invalid" path as far as `Xamarin.Android.Tools.JdkInfo` is concerned, but that was already a possibility when it was within `<ResolveSdks/>`, because `ResolveSdks.JavaSdkPath` wasn't updated if `new JdkInfo(JavaSdkPath)` ever threw.